### PR TITLE
Bring text escaping from Ivory into Mundane

### PR DESCRIPTION
**Pull request by FrontierOtr**

### CODE REVIEW

File: mundane-parse/src/main/scala/com/ambiata/mundane/parse/TextEscaping.scala [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      "delimited"
+    case Escaped =>
+      "escaped"
+  }
+}
+
+object TextEscaping {
+
+  object Delimited extends TextEscaping
+  object Escaped extends TextEscaping
```
Comment (charleso) [inline mundane-parse/src/main/scala/com/ambiata/mundane/parse/TextEscaping.scala]: Hmm, do these make any/much sense outside of Ivory?

Comment (FrontierOtr) [inline mundane-parse/src/main/scala/com/ambiata/mundane/parse/TextEscaping.scala]: Useful for parsing Ivory snapshots and chords?

Comment (markhibberd) [inline mundane-parse/src/main/scala/com/ambiata/mundane/parse/TextEscaping.scala]: I would drop and just have the escaping implementation here. Parsing snapshots/chords is a useful API but not here.

File: mundane-parse/src/main/scala/com/ambiata/mundane/parse/TextEscaping.scala [hunk 2]
```diff
@@ -0,0 +1,86 @@
+package com.ambiata.mundane.parse
+
+import scalaz.Equal
+import scalaz.Scalaz._
+
+object TextEscaping {
+
+  val escapeChar = '\\'
```
Comment (etorreborre) [inline mundane-parse/src/main/scala/com/ambiata/mundane/parse/TextEscaping.scala]: Can we make the `escapeChar` a parameter? Also the `escapeChar` is being used verbatim on L.28.

File: mundane-parse/src/main/scala/com/ambiata/mundane/parse/TextEscaping.scala [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+  def split(delim: Char, s: String): List[String] = {
+    val l = List.newBuilder[String]
+    var i = 0
+    val b = new StringBuilder(s.length)
+    var lastCharWasDelim = false
+    while (i < s.length) {
+      lastCharWasDelim = false
+      val c = s.charAt(i)
+      if (c == '\\') {
```
Comment (etorreborre) [inline mundane-parse/src/main/scala/com/ambiata/mundane/parse/TextEscaping.scala]: And here too.

File: mundane-parse/src/main/scala/com/ambiata/mundane/parse/TextEscaping.scala [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        i += 1
+        if (i < s.length) {
+          val c2 = s.charAt(i)
+          if (c2 == 'n') {
+            b.append('\n')
+          } else if (c2 == 'r') {
+            b.append('\r')
+          } else if (c2 == delim || c2 == '\\') {
```
Comment (etorreborre) [inline mundane-parse/src/main/scala/com/ambiata/mundane/parse/TextEscaping.scala]: Here as well :-)

File: mundane-parse/src/main/scala/com/ambiata/mundane/parse/TextEscaping.scala (hunk 2 above)
Comment (FrontierOtr) [inline mundane-parse/src/main/scala/com/ambiata/mundane/parse/TextEscaping.scala]: I don't think there's many situations where it's not a backslash.

**Comment by etorreborre** [comment]
:+1:

**Pull request merged**